java-client/src/main/java/org/opensearch/client/transport/rest_client/RestClientTransport.java

```diff
@@ -273,7 +274,7 @@ private <ResponseT, ErrorT> ResponseT getHighLevelResponse(
                         // TODO: have the endpoint provide the exception constructor
                         throw new OpenSearchException((ErrorResponse) error);
                     }
-                } catch(MissingRequiredPropertyException errorEx) {
+                } catch(MissingRequiredPropertyException | JsonParsingException errorEx) {
```

Comment:
Should we better off handle the 401/403 separately? It seems like we should not be trying to parse the response entity as JSON knowing it is not the right content type:

```

HTTP/1.1 401 Unauthorized
WWW-Authenticate: Basic realm="OpenSearch Security"
content-type: text/plain; charset=UTF-8
content-length: 12

```